metadata_mapper/mappers/oai/contentdm_mapper.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import itertools
+
+from .oai_mapper import OaiRecord, OaiVernacular
+
+
+class ContentdmRecord(OaiRecord):
+    def UCLDC_map(self):
+        return {
+            "contributor": self.split_and_flatten('contributor'),
+            "creator": self.split_and_flatten('creator'),
+            "spatial": self.map_spatial(),
+            "type": self.map_type(),
+            "language": self.split_and_flatten('language'),
+            "subject": self.map_subject()
+        }
+
+    @staticmethod
+    def identifier_match():
+        return ["cdm/ref"]
+
+    def map_is_shown_at(self):
+        return self.get_matching_identifier(last=True)
+
+    def map_is_shown_by(self):
+        """
+        This was originally a "post-map" function. but really, this is seems like it should be map_is_shown_by()
+
+        Comment from calisphere function:
+        To run post mapping. For this one, is_shown_by needs sourceResource/type
+        """
+        record_type = self.map_type()
+        if not record_type:
+            return
+
+        record_type_list = [record_type] if isinstance(record_type, str) else record_type
+
+        if "sound" in [t.lower() for t in record_type_list]:
+            return None
+
+        return self.get_preview_image_url()
+
+    def map_type(self):
+        """Used by map_is_shown_by() in this class, so cannot be directly added to the mapping"""
+        return self.split_and_flatten('type')
+
+    def map_spatial(self):
+        values = [v for v in self.collate_fields(["coverage", "spatial"]) if v]
+        if not values:
+            return
+
+        split_values = [c.split(';') for c in filter(None, values)]
+
+        return list([s.strip() for s in itertools.chain.from_iterable(split_values)])
+
+    def map_subject(self):
+        subject = self.source_metadata.get('subject')
+        if not subject:
+            return
+
+        return [{'name': v} for v in self.split_and_flatten('subject')]
+
+    def get_preview_image_url(self):
+        """
+        see get_larger_preview_image(), below, as it is mapped to isShownBy as well
+        """
+        larger_preview_image = self.get_larger_preview_image_url()
+        if larger_preview_image:
+            return larger_preview_image
+
+        parts = self.get_identifier_parts()
+        return '/'.join((parts["base_url"], 'utils', 'getthumbnail',
```

Comment:
Could you use an f-string here as well for readability?

Reply:
f-strings didn't exist in python2, so I think that's why this format is so prevalent in the legacy code.